app/presenters/document_collection_presenter.rb

```diff
@@ -40,7 +40,7 @@ def group_document_links(group, group_index)
           }
         },
         metadata: {
-          public_updated_at: Time.zone.parse(link["public_updated_at"]),
+          public_updated_at: link["public_updated_at"]&.then { |time| Time.zone.parse(time) },
```

Comment:
I tried to quickly re-write this as I hadn't seen `then` be given a block before (cool you can do something like this btw), here's my attempt, don't think it's better though so feel free to merge :) `link.fetch("public_updated_at", nil) || Time.zone.parse(link["public_updated_at"])`

Reply:
Sure, `then` was added in Ruby 2.6 as an alias for `yield_self` added in 2.5 so you don't see them much in the wild. It's a really useful method though: https://mlomnicki.com/yield-self-in-ruby-25/

Your suggestion there won't work as the right side of the `||` will be called when the left side returns nil. The alternative approach to this is through the ternary operator: `link["public_updated_at"] ? Time.zone.parse(link["public_updated_at"]) : nil,` or a normal conditional: `Time.zone.parse(link["public_updated_at"]) if link["public_updated_at"]` - though I always think normal conditionals are a bit confusing inside a hash.

Reply:
Oh yeah of course, my bad! Personally I like `Time.zone.parse(link["public_updated_at"]) if link["public_updated_at"]` as it reads quite nicely to me although it doesn't explicitly return nil.

Reply:
I'm not a big fan of having that when it's within a hash. For instance this would be:

```
metadata: {
  public_updated_at: Time.zone.parse(link["public_updated_at"]) if link["public_updated_at"],
}
```

and I read that as it being ambiguous exactly what on the line the if applies to. 

I'm going to roll with the `then`